### PR TITLE
230 create uiconfig service

### LINKED ISSUE #230: 3c UIConfig: create new UIConfig service
While we don't currently have a use case for most of the CRUD actions on UIConfig, we do have a need for the UPDATE action!

- [ ] Create a new service for UIConfig
- [ ] Write test coverage for an `update` function. It should, at a minimum, completely cover the way `dashboardStreams` are updated via device (do not remove how device dashboard's update themselves through device yet). 
- [ ] Create an `update` lambda controller for UIConfig
- [ ] pair with viktor on the steps required to deploy this new service

### PR DESCRIPTION
**Pull request by O-Susanoo**
closes #230

### CODE REVIEW

File: services/uiconfig/test/event-generators/createPutConfigEvent.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const createPutConfigEvent = (userId, deviceId, uiConfig, requestingUserId) => ({
```
Comment (vdiezel) [inline services/uiconfig/test/event-generators/createPutConfigEvent.js]: can you please also add  `httpMethod: 'PUT` to this event. Some middleware determines based on the presens of this key if the request is a HTTP event or not.

File: services/uiconfig/test/put.test.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    it('returns 422 and a msg when the uiconfig is missing', async () => {
+      const { res, body } = await firePutUIConfigEvent(uuidv4(), uuidv4())
+
+      expect(res.statusCode).toBe(422)
+      assertContainsAll(body.message, [ 'configs', 'Validation' ])
+    })
+  })
+
+  describe('when the requesting user is not the same as userId', () => {
+    it('returns 403 and a msg when the deviceId is missing', async () => {
```
Comment (vdiezel) [inline services/uiconfig/test/put.test.js]: You forgot to update this text :japanese_ogre:

File: services/uiconfig/src/uiconfig/put.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const requestData = {
+    userId: event.pathParameters.userId,
+    deviceId: event.pathParameters.deviceId,
+    configs: event.body.configs,
+  }
+  const { requestingUserId } = event
+
+  validatePutUIConfig(requestData)
+
+  assert(() => isMe(requestingUserId, requestData.userId))
```
Comment (DanielSeehausen) [inline services/uiconfig/src/uiconfig/put.js]: 👍

Comment (DanielSeehausen) [reply services/uiconfig/src/uiconfig/put.js]: in the controller nice

File: services/uiconfig/test/put.test.js (hunk 2 above)
Comment (O-Susanoo) [reply services/uiconfig/test/put.test.js]: Sorry plz don't unleash the devil on me

File: services/uiconfig/test/event-generators/createPutConfigEvent.js (hunk 1 above)
Comment (O-Susanoo) [reply services/uiconfig/test/event-generators/createPutConfigEvent.js]: Done , thank you for the laser eye :+1:

**Pull request merged**